core/domain/platform_parameter_registry.py

```diff
@@ -0,0 +1,300 @@
+            PlatformParameter. The created feature flag.
+        """
+        return cls.create_platform_parameter(
+            name, description, 'bool', is_feature=True, feature_stage=stage)
+
+    @classmethod
+    def init_platform_parameter(cls, name, instance):
+        """Initializes parameter_registry with keys as the parameter names and
+        values as instances of the specified parameter.
+
+        Args:
+            name: str. The name of the platform parameter.
+            instance: PlatformParameter. The instance of the platform parameter.
+        """
+        if cls.parameter_registry.get(name):
+            raise Exception('Parameter with name %s already exists.' % name)
+        cls.parameter_registry[name] = instance
+
+    @classmethod
+    def get_platform_parameter(cls, name):
+        """Returns the instance of the specified name of the platform
+        parameter.
+
+        Args:
+            name: str. The name of the platform parameter.
+
+        Returns:
+            PlatformParameter. The instance of the specified platform
+            parameter.
+        """
+        parameter_from_cache = cls.load_platform_parameter_from_memcache(
+            name)
+        if parameter_from_cache is not None:
+            return parameter_from_cache
+
+        parameter = None
+
+        parameter_from_storage = cls.load_platform_parameter_from_storage(
+            name)
+        if parameter_from_storage is not None:
+            parameter = parameter_from_storage
+        elif name in cls.parameter_registry:
+            parameter = cls.parameter_registry[name]
+        else:
+            raise Exception('Platform parameter not found: %s.' % name)
+
+        memcache_key = param_domain.PlatformParameter.get_memcache_key(name)
+        memcache_services.set_multi({
+            memcache_key: parameter,
+        })
+        return parameter
+
+    @classmethod
+    def update_platform_parameter(
+            cls, name, committer_id, commit_message, new_rule_dicts):
+        """Updates the platform parameter with new rules.
+
+        Args:
+            name: str. The name of the platform parameter to update.
+            committer_id: str. ID of the committer.
+            commit_message: str. The commit message.
+            new_rule_dicts: list(dist). A list of dict mappings of all fields
+                of PlatformParameterRule object.
+        """
+        param = cls.get_platform_parameter(name)
+
+        # Create a temporary param instance with new rules for validation,
+        # if the new rules are invalid, an exception will be raised in
+        # validate() method.
+        param_dict = param.to_dict()
+        param_dict['rules'] = new_rule_dicts
+        updated_param = param.from_dict(param_dict)
+        updated_param.validate()
+
+        # Set value in datastore.
+        model_instance = config_models.PlatformParameterModel.get(
+            param.name, strict=False)
+        if model_instance is None:
```

Comment:
If this is a new instance, I don't think we also want to perform a commit updating it right away. I'd expect this method to create or update an entity, not potentially do both.

Reply:
I was taking config properties as reference:
https://github.com/oppia/oppia/blob/04570fb22780d9a22d94e33916a1d0e8d17c2a6d/core/domain/config_domain.py#L230-L251

I think we do it this way to make the storage process transparent to `Registry.update_platform_parameter` calls -- the update is persistent in storage, and if there isn't a previously created model in storage, we create one.

Also I refactored this part out to another method `_to_platform_parameter_model(cls, param)` so that the flow of `update_platform_parameter` might be clearer.